compiler/modules/bir/bir.bal

```diff
@@ -127,17 +127,41 @@ public function lastInsnRef(BasicBlock bb) returns InsnRef {
     return { block: bb.label, index: bb.insns.length() - 1 };
 }
 
-public type Register readonly & record {|
+public distinct readonly class Register {
```

Comment:
No. Leave it as a record.

Just change varName to have type `PositionString?`

```
public type PositionString readonly & [Position, string];
```

Subtyping is only relevant to codeGen. So in that module create a structural subtype

```
type PositionRegister record {|
  *bir:Register;
  PositionString varName;
|};
```

Reply:
I considered doing this but if I did 
```
public type PositionString readonly & [Position, string];

public type Register readonly & record {|
    int number;
    SemType semType;
    PositionString? varName;
|};

public type PositionRegister readonly & record {|
    *Register;
    PositionString varName;
|};
```
I am getting an error at "*Register" saying "incompatible types: 'Register' is not a record. We can't also do the workaround we have used in `ast.bal` by replacing "*Register" with 
```
int number;
SemType semType;
```
because this don't allow me cast from `Register` to `PositionRegister`

Reply:
It's probably the `readonly` that is causing the problem.

In any case, it's a complete non-starter to use a class here. So get rid of that first. If you can't make the subtype work, it's no big deal.